File: src-tauri/src/dto/bookshelf.rs
use super::{comic::ComicSummary, novel::NovelSummary};
use serde::Serialize;

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub(crate) struct NovelBookshelfEntry {
    pub book: NovelSummary,
    pub added_at: String,
    pub progress: Option<i64>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub(crate) struct ComicBookshelfEntry {
    pub comic: ComicSummary,
    pub added_at: String,
    pub progress: Option<i64>,
}

File: src/domain/bookshelf.ts
import type { NovelSummary } from "./novel";
import type { ComicSummary } from "./comic";

/** A novel shelf record returned by the application boundary. */
export interface NovelBookshelfEntry {
  book: NovelSummary;
  addedAt: string;
  progress: number | null;
}

/** A comic shelf record returned by the application boundary. */
export interface ComicBookshelfEntry {
  comic: ComicSummary;
  addedAt: string;
  progress: number | null;
}
